readthedocs/projects/forms.py

```diff
@@ -272,6 +273,69 @@ def __init__(self, *args, **kwargs):
         else:
             self.fields['default_version'].widget.attrs['readonly'] = True
 
+        self.setup_external_builds_option()
+
+    def setup_external_builds_option(self):
+        """Disable the external builds option if the project doesn't meet the requirements."""
+        msg = None
+        url = reverse("projects_integrations", args=[self.instance.slug])
+        integrations = list(self.instance.integrations.all())
+        if not self.has_supported_integration(integrations):
```

Comment:
I found this logic of `if not ... and elif not` pretty hard to read since both are negated. Is it possible to change it to handle the valid and simple case first, something like this:

```
if has_supported_integration and can_build_external_versions:
  return

if not can_build_external_versions:
  msg = _("You need the event...")

if not has_supported_integration:
  msg = _("To build from pull request you...")
```